### ISSUE TITLE
Change MONO pre-processor flags into runtime checks => Mono can use Nuget Packages

### ISSUE THREAD

**Issue by damageboy** [NLog/NLog#1349]
**Type** (choose one):

Bug

**NLog version**: (e.g. 4.2.3)
Any nlog version

**Platform**: 
Mono 4 

Hi,
As I'm trying to fix a dead-lock that happens in Mono (only), I've come across different commits in the source code were special behavior was put in place for Mono support, for example:
https://github.com/NLog/NLog/commit/e58867547be4fc1ff9a1c484d723b3ae12d7333e

As can be clearly seen from the actual commit code, fix involved using a `#if MONO` pre-processor condition.

While this was perhaps a correct solution for that time (2011, in the above case), it's obviously doesn't play nice with how people consume NLog in 2016.
Namely, NLog is mostly distributed throughout NuGet, where there is no way do provide different binaries for Standard .NET (MS.NET) and Mono (It's also probably important to make note that this is a by-design decision taken by the NuGet/Mono people).

As such, it looks like it would be more correct to change the Mono checks, specifically, to become runtime checks.

There is a well documented way of detecting Mono @ Runtime:

``` c#
public static bool IsRunningOnMono ()
{
    return Type.GetType ("Mono.Runtime") != null;
}
```

I think that the NLog source, as it has to be compiled once into a single .DLL for MS.NET + Mono should transition into using runtime checks to support conditional execution in Mono...

**Comment by 304NotModified** [comment]
This would be great to have. 

We consider this as an feature, as the Mono support was always source distributed and not binary (nuget)

**Comment by 304NotModified** [comment]
Fixed by #1893 & #1833
